blazar_dashboard/api/client.py

```diff
@@ -272,55 +302,37 @@ def host_reservation_dict(reservation, resource_id):
     return compute_hosts, list(chain(*host_reservations))
 
 
-def network_list(request):
-    """Return a list of networks available for reservation"""
-    # TODO > USE CLIENT
-    sql = '''\
-    SELECT
-        physical_network,
-        segment_id
-    FROM
-        network_segments
-    '''
-    cursor = get_cursor_for_request(request)
-    cursor.execute(sql)
-    networks = dictfetchall(cursor)
+def network_reservation_calendar(request):
+    """Return a list of all scheduled network leases."""
 
-    return networks
+    def network2dict(n):
+        return dict(
+            network_type=n.network_type, physical_network=n.physical_network,
+            segment_id=n.segment_id)
 
+    networks_by_id = {n.id: n for n in network_list(request)}
 
-def network_reservation_calendar(request):
-    """Return a list of all scheduled leases."""
-    cursor = get_cursor_for_request(request)
-    sql = '''\
-    SELECT
-        l.name,
-        l.project_id,
-        l.start_date,
-        l.end_date,
-        r.id,
-        r.status,
-        n.segment_id
-    FROM
-        network_allocations na
-        JOIN network_segments n ON n.id = na.network_id
-        JOIN reservations r ON r.id = na.reservation_id
-        JOIN leases l ON l.id = r.lease_id
-    WHERE
-        r.deleted IS NULL
-        AND na.deleted IS NULL
-    ORDER BY
-        start_date,
-        project_id;
-    '''
-    cursor.execute(sql)
-    network_reservations = dictfetchall(cursor)
+    def network_reservation_dict(reservation, resource_id):
+        network_reservation = dict(
+            name=reservation.get('name'),
+            project_id=reservation.get('project_id'),
+            start_date=_parse_api_datestr(reservation['start_date']),
+            end_date=_parse_api_datestr(reservation['end_date']),
+            id=reservation['id'],
+            status=reservation.get('status'),
+            segment_id=networks_by_id[resource_id].segment_id)
```

Comment:
Why is segment_id returned here as well?

Reply:
That's what the calendar uses as the label. The original db call grabbed it from the database. We can either apply it here or refactor the javascript for the calendar.